src/Matcher.php

```diff
@@ -24,14 +38,27 @@ public function getMatches($password, array $userInputs = [])
         foreach ($this->getMatchers() as $matcher) {
             $matched = $matcher::match($password, $userInputs);
             if (is_array($matched) && !empty($matched)) {
-                $matches = array_merge($matches, $matched);
+                $matches[] = $matched;
             }
         }
 
+        $matches = array_merge([], ...$matches);
         self::usortStable($matches, [$this, 'compareMatches']);
+
         return $matches;
     }
 
+    public function addMatcher(string $className)
+    {
+        if (!in_array(MatchInterface::class, class_implements($className), true)) {
```

Comment:
This does the same but is easier to read, I think:

```suggestion
        if (!is_a($className, MatchInterface::class, true)) {
```